src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -271,13 +278,40 @@ public static Complex Acos(Complex value)
 
         public static Complex Tan(Complex value)
         {
-            return (Sin(value) / Cos(value));
+            // tan z = sin z / cos z, but to avoid unnecessary repeated trig computations, use
+            //   tan z = (sin(2x) + i sinh(2y)) / (cos(2x) + cosh(2y))
+            // (see Abramowitz & Stegun 4.3.57 or derive by hand), and compute trig functions here.
+
+            // Even this can only be used for |y| not-too-big, beacuse sinh and cosh (correctly) overflow
```

Comment:
The wording is a little awkward. Consider:

> The approach does not work for moderatlely-large values of |y|, because sinh and cosh will overflow. In that case, divide through by ...